Replace IRIs

garemoko opened the pull request:
As this is in the public ADL github and listed in the CoP Google Group, there's a reasonable chance somebody could look at this as a guide for the semantics of statement structure, which I understand is not the intention of this project. Instead people should look to existing implementations as documented in the [Registry](https://registry.tincanapi.com/#profile/19) and the work of the [Video CoP](http://www.adlnet.gov/tla/experience-api/xapi-cop-directory/xapi-cop-directory-list-view/entry/videos/). 

This PR adds to the readme to make clear that this is not a statement structure guide and replaces a number of the IRIs used. 

Note that this PR does not make this project fully conformant to the Registry recipe, as not all data and events are represented.

garemoko committed:
Add activity type

garemoko committed:
verb and extension IRIs

garemoko committed:
Make clear this isn't a model for statement semantics

brianjmiller commented:
It would make more sense to leverage the existing video recipe library itself rather than hard coding this stuff here, so that if there are improvements or new versions of the recipe added to that library in the future this project gets them automatically. The recipe library is video player agnostic.

garemoko commented:
Hi! Any word on accepting this PR. Is it worth my time solving the merge conflict? I don't plan to put the time in to have this use the recipe library though I agree this would be good to do.

cawerkenthin commented:
Hey Andrew,
Am I wrong that your verbs require "id":?, like
"verb": { "id": "http://id.tincanapi.com/verb/paused", "display": { "en-US": "paused" } }

garemoko commented:
You are quite right. It's been a while since I created this and I'm not sure whether or not the wrapper creates the object from the IRI string. 

Either way, I agree that including the full object with display language maps would be an improvement. 

If @ty- confirms that it's worthwhile me putting the work in (I.e. this will be merged) then I'll make that change when resolving the conflict.

cawerkenthin commented:
I am using your changes; hopefully @ty- would merge

garemoko commented:
How are you using it? Be aware that even with these changes, this project doesn't fully conform to the [video recipe](https://registry.tincanapi.com/#profile/19). 

If you do make changes on top of my changes, feel free to make a PR to [my branch](https://github.com/garemoko/xapi-youtube/tree/garemoko-IRIs-patch) to be included in this PR.

cawerkenthin commented:
I am using to write xAPI statements from our LMS when you view a YouTube video. I'll take a look at the recipe.

cawerkenthin committed:
Match recipe

I matched the recipe as I understood it.  I don't believe there are more events to respond to as the YouTube API does not have a "Skip" for example, which is too bad.  

I also moved "extensions" to the context object in a couple of places, instead of the results object.

cawerkenthin committed:
Update xapi-youtube-statements.js

garemoko committed:
Merge pull request #1 from cawerkenthin/patch-1

Match recipe

garemoko commented:
Thanks for that PR @cawerkenthin 

Another note that might be useful to you: the recipe library @brianjmiller mentions above can be found here: https://github.com/RusticiSoftware/TinCanRecipes/blob/master/video/src/recipes-video.js

With an example of use here: https://github.com/RusticiSoftware/TinCanRecipes/blob/master/site/src/views/app/video.js

cawerkenthin committed:
Track skipped

Need to add the following to your HTML:

var playerPreviousState;
var playerPreviousTime;

cawerkenthin committed:
Update xapi-youtube-statements.js

garemoko committed:
Merge pull request #2 from cawerkenthin/patch-2

Track skipped

garemoko committed:
Fix conflict

garemoko commented:
@cawerkenthin I've resolved the conflict as best I can but it was a fairly signifiant conflict and it's possible I've created a Frankenstein's monster that has bits of two forks and doesn't actually work. Please test!

creighton commented:
obe
